core/stackref_test.go

```diff
@@ -55,4 +64,90 @@ func TestStackRef(t *testing.T) {
 	if !testutils.EqualInt64Slice(observedVals, expectedVals) {
 		t.Error("iterating through the stackref should return the elements in the order they were pushed. Expected:", expectedVals, ", got: ", observedVals)
 	}
+
+	sr2 := core.NewStackRef()
+	obj1 := core.NewObj(10, -1, core.OBJ_TYPE_STRING, core.OBJ_ENCODING_INT)
+	core.Put("key1", obj1)
+	sr2.Push("key1")
+
+	if key, err := sr2.Pop(); err != nil || key.Obj != core.Get("key1") {
+		t.Error("popping non-expired key from stackref should pop element")
+
+	}
+
+	keys := []struct {
+		key           string
+		value         int
+		expDurationMs int64
+	}{
+		{"key0", 0, 0},     // expired key
+		{"key1", 10, 0},    // expired key
+		{"key2", 20, -1},   // non-expired key
+		{"key3", 30, 0},    // expired key
+		{"key4", 40, 0},    // expired key
+		{"key5", 50, 0},    // expired key
+		{"key6", 60, -1},   // non-expired key
+		{"key7", 70, 0},    // expired key
+		{"key8", 80, -1},   // non-expired key
+		{"key9", 90, -1},   // non-expired key
+		{"key10", 100, -1}, // non-expired key
+	}
+
+	for _, k := range keys {
+		CreateAndPushKeyToStack(sr2, k.key, k.value, k.expDurationMs)
+	}
+
+	expectedVals = []int64{100, 90, 80}
+	observedVals = make([]int64, 0, 3)
+
+	for i := 0; i < len(expectedVals); i++ {
+		key, err := sr2.Pop()
+		if err != nil {
+			t.Errorf("unexpected error: %v", err)
+		}
+		if key == nil {
+			t.Errorf("expected key, got nil")
+		} else {
+			observedVals = append(observedVals, int64(key.Obj.Value.(int)))
+		}
+	}
+
+	if !testutils.EqualInt64Slice(observedVals, expectedVals) {
+		t.Errorf("expected %v, got %v", expectedVals, observedVals)
+	}
+
+	if key, err := sr2.Pop(); err != nil || int64(key.Obj.Value.(int)) != 60 {
+		t.Errorf("popping an expired key followed by a non-expired key should work")
+	}
+
+	if key, err := sr2.Pop(); err != nil || int64(key.Obj.Value.(int)) != 20 {
+		t.Errorf("popping multiple expired keys followed by a non-expired key should work")
+	}
+
+	if _, err := sr2.Pop(); err != core.ErrStackEmpty {
+		t.Errorf("popping all expired keys till stack becomes empty should work")
+	}
+
+}
+
+func BenchmarkRemoveLargeNumberOfExpiredKeys(b *testing.B) {
+	for _, v := range benchmarkDataSizes {
+		config.KeysLimit = 20000000 // Set a high limit for benchmarking
+		core.WatchChannel = make(chan core.WatchEvent, config.KeysLimit)
+
+		b.Run(fmt.Sprintf("keys_%d", v), func(b *testing.B) {
+			for i := 0; i < b.N; i++ {
+				b.StopTimer()
+				sr := core.NewStackRef()
+				CreateAndPushKeyToStack(sr, "initialKey", 0, -1)
+				for j := 1; j < v; j++ {
+					CreateAndPushKeyToStack(sr, strconv.Itoa(j), j, 0)
+				}
+				b.StartTimer()
+				if _, err := sr.Pop(); err == core.ErrStackEmpty {
```

Comment:
Ideally we shouldn't ever hit the StackEmpty error, right? Since we only pop once we should pop the non-expired key.

In what case will we run into the ErrStackEmpty condition?

Reply:
yes we will not hit that case but if it was failing in linting in pipeline 
if i write
`sr.Pop()`
because it has return values

should i just do
_,_ = sr.Pop() instead of above ?

Reply:
Okay, Sure